demo/src/main/java/com/novoda/storagepathfinder/demo/AssetCloner.java

```diff
@@ -0,0 +1,74 @@
+package com.novoda.storagepathfinder.demo;
+
+import android.content.res.AssetManager;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.Executor;
+
+class AssetCloner {
+
+    private static final int BUFFER_SIZE = 1024;
+    private static final byte[] BUFFER = new byte[BUFFER_SIZE];
+
+    private final AssetManager assetManager;
+    private final Executor executor;
+
+    AssetCloner(AssetManager assetManager, Executor executor) {
+        this.assetManager = assetManager;
+        this.executor = executor;
+    }
+
+    public void cloneAsset(String assetName, String pathToCloneTo) {
```

Comment:
you might want a callback for when its finished?

Reply:
I avoided this for now because I'd need to hook it back into the adapter, to show next to the root that it was called with. 

I'd rather do that when I come back and do more with the demo, I have plans 😄